### PR TITLE
Set numerous fields in VKDevice::createGraphicsPipelineState using the information in GraphicsPipelineStateDesc

### PR DESCRIPTION
**Pull request by lucy96chen**
Changes:
- Numerous fields in the pipeline descs filled in inside `VKDevice::createGraphicsPipelineState()` now take their values where appropriate from the corresponding fields in the `GraphicsPipelineStateDesc` provided in the arguments instead of being hard-coded
- Small naming convention fix to `sampleCount` added alongside `resolveResource`

Note: There are a number of TODOs for some of the fields as either gfx does not currently support them or Vulkan and D3D12 treat them differently. These will be addressed at a later date.

@csyonghe

### CODE REVIEW

File: tools/gfx/vulkan/render-vk.cpp [hunk 1]
```diff
@@ -8220,6 +8223,118 @@ Result VKDevice::createMutableRootShaderObject(
...
+    case CullMode::Back:        return VK_CULL_MODE_BACK_BIT;
+    default:
+        assert(!"Unsupported cull mode");
+        return VK_CULL_MODE_NONE;
+    }
+}
+
+VkFrontFace translateFrontFaceMode(FrontFaceMode frontFaceMode)
+{
+    // TODO: May need to be reversed due to the viewport flip
```
Comment (csyonghe) [inline tools/gfx/vulkan/render-vk.cpp:8258]: That's right, this needs to be reversed.

Comment (lucy96chen) [reply tools/gfx/vulkan/render-vk.cpp:8258]: Fixed locally, but doesn't seem to be the cause of nothing being drawn in some of those tests.

File: tools/gfx/vulkan/render-vk.cpp [hunk 2]
```diff
@@ -8305,43 +8399,87 @@ Result VKDevice::createGraphicsPipelineState(const GraphicsPipelineStateDesc& in
...
+    }
+    
     VkPipelineColorBlendStateCreateInfo colorBlending = {};
     colorBlending.sType = VK_STRUCTURE_TYPE_PIPELINE_COLOR_BLEND_STATE_CREATE_INFO;
-    colorBlending.logicOpEnable = VK_FALSE;
+    colorBlending.logicOpEnable = VK_FALSE; // TODO: D3D12 has per attachment logic op (and both have way more than one op)
     colorBlending.logicOp = VK_LOGIC_OP_COPY;
-    colorBlending.attachmentCount = 1;
-    colorBlending.pAttachments = &colorBlendAttachment;
+    colorBlending.attachmentCount = (targetCount == 0) ? 1 : targetCount;
```
Comment (csyonghe) [inline tools/gfx/vulkan/render-vk.cpp:8466]: This can just be (uint32_t)colorBlendAttachments.getCount()

**Review by csyonghe — APPROVED** [review]

File: tools/gfx/vulkan/render-vk.cpp (hunk 2 above)
Comment (lucy96chen) [reply tools/gfx/vulkan/render-vk.cpp:8466]: Fixed locally, will be in next PR

**Pull request merged**